### PR TITLE
CNTRLPLANE-4008: enable hypershiftlinter and fix test naming

### PR DESCRIPTION
**Pull request by bryan-cox**
## What this PR does / why we need it:
Enables the hypershiftlinter golangci-lint plugin (added in #9237) and fixes all existing test naming violations across the codebase to match TESTING.md conventions (`When ..., it should ...`).

Also fixes `make test-changed` to exclude the `api/` submodule (which has its own `go.mod` and cannot be listed from the root module).

## Which issue(s) this PR fixes:
Fixes [CNTRLPLANE-4008](https://redhat.atlassian.net/browse/CNTRLPLANE-4008)

## Special notes for your reviewer:
Depends on #9237 merging first so the `hypershiftlinter.so` plugin binary exists.

## Checklist:
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs.
- [x] This change includes unit tests.

## Summary by CodeRabbit

* **Chores**
  * Enabled an additional linting check and configured it for the project.
  * Updated lint commands to automatically prepare the required linting component.

* **Tests**
  * Improved test names and descriptions across cluster, infrastructure, controller, networking, security, and utility coverage.
  * Standardized test naming conventions and clarified scenarios and expected outcomes without changing existing behavior.
  * Added coverage for cluster lookup retries, Azure destroy handling, and node selector parsing edge cases.

### CODE REVIEW

File: hack/tools/hypershiftlinter/analyzers/testcasename/testcasename.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	"github.com/openshift/hypershift/hack/tools/hypershiftlinter/analyzers/pathutil"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name: "testcasename",
+	Doc:  "checks that test case name fields match \"When <condition>, it should <expected behavior>\"",
+	Run:  run,
+}
+
+var namePattern = regexp.MustCompile(`(?i)^when .+,? it should .+$`)
```
[the remark this answers is not in the document]
Comment (bryan-cox) [reply hack/tools/hypershiftlinter/analyzers/testcasename/testcasename.go]: Done. The analyzer now requires the exact uppercase `When <condition>, it should <behavior>` form, with focused coverage for lowercase rejection.

---
*AI-assisted response via Claude Code*

File: etcd-backup/etcdbackup_test.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ func TestMapToTags(t *testing.T) {
 		validateFunc func(t *testing.T, result string)
 	}{
 		{
-			name: "When tags are provided it should URL-encode them correctly",
+			name: "When tags are provided, it should URL-encode them correctly",
```
[the remark this answers is not in the document]
Comment (bryan-cox) [reply etcd-backup/etcdbackup_test.go:49]: Done. All governed test cases were migrated to the exact required format, and `make lint` now passes with the analyzer enabled.

---
*AI-assisted response via Claude Code*

File: hack/tools/hypershiftlinter/plugin.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	"github.com/openshift/hypershift/hack/tools/hypershiftlinter/analyzers/vacuouspass"
+)
+
+type Settings struct {
+	Analyzers *AnalyzerSettings `json:"analyzers"`
+}
+
+type AnalyzerSettings struct {
+	Enable []string `json:"enable"`
+}
```
[the remark this answers is not in the document]
Comment (bryan-cox) [reply hack/tools/hypershiftlinter/plugin.go]: Done. The settings types and analyzer registry are now unexported; `BuildAnalyzers` remains the only required exported plugin API.

---
*AI-assisted response via Claude Code*

File: .github/workflows/test-linter.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+name: Linter Unit Tests
+
+on:
+  pull_request:
+    branches:
+      - main
+      - release-4.22
+    paths:
+      - 'hack/tools/hypershiftlinter/**'
+
```
[the remark this answers is not in the document]
Comment (bryan-cox) [reply .github/workflows/test-linter.yaml:16]: Done. The workflow now triggers for hack/tools/go.mod, hack/tools/go.sum, vendor changes, the Makefile, and both linter workflow files.

---
*AI-assisted response via Claude Code*

File: Makefile
Comment (cblecker) [inline Makefile]:
```suggestion
	cd $(TOOLS_DIR); CGO_ENABLED=1 $(GO) build -a -buildmode=plugin -o $(HYPERSHIFTLINTER_PLUGIN) ./hypershiftlinter/cmd/plugin
```
plugin mode requires CGO.. this ensures that it is turned on for this build

**Review by cblecker — COMMENTED** [review]
one functional nit, otherwise lgtm

**Review by cblecker — APPROVED** [review]

**Comment by bryan-cox** [comment]
/verified by [Lint / lint / Lint (pull_request)](https://github.com/openshift/hypershift/actions/runs/32289040936/job/96185339294?pr=9271)

**Review by everettraven — COMMENTED** [review]
/approve for api changes
